windows-rdp/main.tf

```diff
@@ -0,0 +1,177 @@
+      Install-Module -Name $moduleName -RequiredVersion $moduleVersion -Force
+    }
+    catch {
+      # If the first command failed, assume that we're on GCP and run
+      # Install-Module only
+      Install-Module -Name $moduleName -RequiredVersion $moduleVersion -Force
+    }
+
+    # Construct the module path for system-wide installation
+    $moduleBasePath = "C:\Windows\system32\config\systemprofile\Documents\PowerShell\Modules\$moduleName\$moduleVersion"
+    $modulePath = Join-Path -Path $moduleBasePath -ChildPath "$moduleName.psd1"
+
+    # Import the module using the full path
+    Import-Module $modulePath
+    Install-DGatewayPackage
+
+    # Configure Devolutions Gateway
+    $Hostname = "localhost"
+    $HttpListener = New-DGatewayListener 'http://*:7171' 'http://*:7171'
+    $WebApp = New-DGatewayWebAppConfig -Enabled $true -Authentication None
+    $ConfigParams = @{
+      Hostname = $Hostname
+      Listeners = @($HttpListener)
+      WebApp = $WebApp
+    }
+    Set-DGatewayConfig @ConfigParams
+    New-DGatewayProvisionerKeyPair -Force
+
+    # Configure and start the Windows service
+    Set-Service 'DevolutionsGateway' -StartupType 'Automatic'
+    Start-Service 'DevolutionsGateway'
+  }
+
+  function Patch-Devolutions-HTML {
+    $root = "C:\Program Files\Devolutions\Gateway\webapp\client"
+    $devolutionsHtml = "$root\index.html"
+    $patch = '<script defer id="coder-patch" src="coder.js"></script>'
+    
+    # Always copy the file in case we change it.
+    @'
+${templatefile("${path.module}/devolutions-patch.js", {
+  CODER_USERNAME : var.admin_username,
+  CODER_PASSWORD : var.admin_password,
+})}
+'@ | Set-Content "$root\coder.js"
+
+    # Only inject the src if we have not before.
+    $isPatched = Select-String -Path "$devolutionsHtml" -Pattern "$patch" -SimpleMatch
+    if ($isPatched -eq $null) {
+      (Get-Content $devolutionsHtml).Replace('</app-root>', "</app-root>$patch") | Set-Content $devolutionsHtml
+    }
+  }
+
+  Set-AdminPassword -adminPassword "${var.admin_password}"
+  Configure-RDP
+  Install-DevolutionsGateway
+  Patch-Devolutions-HTML
+
+  EOF
+
+run_on_start = true
+}
+
+resource "coder_app" "windows-rdp" {
+  agent_id     = var.agent_id
+  slug         = "web-rdp"
+  display_name = "Web RDP"
+  url          = "http://localhost:7171"
+  icon         = "https://svgur.com/i/158F.svg"
+  subdomain    = true
+
+  healthcheck {
+    url       = "http://localhost:7171"
+    interval  = 5
+    threshold = 15
+  }
+}
+
+resource "coder_app" "rdp-docs" {
+  agent_id     = var.agent_id
+  display_name = "Local RDP"
+  slug         = "rdp-docs"
+  icon         = "https://raw.githubusercontent.com/matifali/logos/main/windows.svg"
+  url          = "https://coder.com/docs/v2/latest/ides/remote-desktops#rdp-desktop"
+  external     = true
+}
+
+# For some reason this is not rendering, commented out for now
+# resource "coder_metadata" "rdp_details" {
```

Comment:
Is there an issue to follow-up on this?
I don't see any other module defining a `coder_metadata` resource.

Reply:
Oh, actually I forgot about this. This was a holdover from Ben's initial prototype
@bpmct Is it safe to delete all this?

Reply:
Yeah, it's safe to delete. Personally, I would love to render it but I couldn't seem to get it working